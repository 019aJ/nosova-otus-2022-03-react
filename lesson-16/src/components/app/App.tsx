import { useEffect, useState } from "react"
import { AuthContext } from "../../context"
import { BrowserRouter } from "react-router-dom"
import { AppRouter } from "../approutes/AppRouter"
import { Navbar } from "../navbar/Navbar"

interface AppProps {}

export const App: React.FC<AppProps> = ({}) => {
  const [userName, setUserName] = useState<string | null>(null)

  useEffect(() => {
    setUserName(localStorage.getItem("auth"))
  }, [])
  return (
    <AuthContext.Provider
      value={{
        name: userName,
        saveName: setUserName,
      }}
    >
      <BrowserRouter>
        <Navbar />
        <AppRouter />
      </BrowserRouter>
    </AuthContext.Provider>
  )
}
