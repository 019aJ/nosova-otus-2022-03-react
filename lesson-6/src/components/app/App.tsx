import Field from "../field/Field"
import { Component } from "react"

interface AppProps {
  cellCount?: number
}

export default class App extends Component<AppProps> {
  constructor(props: AppProps) {
    super(props)
  }
  render() {
    return (
      <div className="App">
        <Field cellCount={this.props.cellCount || 0}></Field>
      </div>
    )
  }
}
