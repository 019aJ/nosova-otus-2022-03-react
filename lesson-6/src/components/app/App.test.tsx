import { render, screen } from '@testing-library/react';
import App from './App';

describe("App snapshot tests", () => {
  it("base snapshot", () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  it("some data snapshot", () => {
    const { container } = render(
      <App cellCount={12} />
    )
    expect(container).toMatchSnapshot()
  })
})
