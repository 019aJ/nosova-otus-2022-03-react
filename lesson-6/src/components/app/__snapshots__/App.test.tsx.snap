// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot tests base snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      style="background-color: yellow; width: 500px; height: 500px;"
    >
      Field
    </div>
  </div>
</div>
`;

exports[`App snapshot tests some data snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <div
      style="background-color: yellow; width: 500px; height: 500px;"
    >
      <div
        style="white-space: nowrap;"
      >
        <div
          style="display: inline-block;"
        >
          <div
            id="1"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="2"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="3"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="4"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
      </div>
      <div
        style="white-space: nowrap;"
      >
        <div
          style="display: inline-block;"
        >
          <div
            id="5"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="6"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="7"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="8"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
      </div>
      <div
        style="white-space: nowrap;"
      >
        <div
          style="display: inline-block;"
        >
          <div
            id="9"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="10"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="11"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
        <div
          style="display: inline-block;"
        >
          <div
            id="12"
            style="background-color: pink; width: 123px; height: 123px; border: 1px solid black;"
          >
            <div
              style="text-align: center; position: relative; top: 30%;"
            >
              click me
            </div>
          </div>
        </div>
      </div>
      <div
        style="white-space: nowrap;"
      />
    </div>
  </div>
</div>
`;
